fatpack/endurance.py

```diff
@@ -411,3 +506,187 @@ def get_stress(self, N):
 if __name__ == "__main__":
     import doctest
     doctest.testmod()
+
+
+class QuadLinearEnduranceCurve(AbstractEnduranceCurve):
+    """Define a quadlinear endurance curve.
+
+            ^                log N - log S
+            |
+            |*
+            | *  m1
+            |  *- -+
+            |   *  .
+     S   Sc +- - * . 1
+     t      |    .*.
+     r      |    . *
+     e      |    .  *
+     s   Sd +- - - - *
+     s      |    .    . *      m2
+            |    .    .    * - - - -+
+            |    .    .       *     . 1
+            |    .    .          *  .
+            |    .    .             *
+            |    .    .                *
+         Sl +- - - - - - - - - - - - - - -*
+                 .    .                 .  *
+            |    .    .                 .   *  m3
+            |    .    .                 .    *- -+
+            |    .    .                 .     *  . 1
+     S           .    .                 .      * .
+     t      |    .    .                 .       *.
+     r      |    .    .                 .        *
+     e      |    .    .                 .         *
+     s   Sk +- - - - - - - - - - - - - - - - - - - *
+     s      |    .    .                 .          . *      m4
+            |    .    .                 .          .    * - - - -+
+            |    .    .                 .          .       *     . 1
+            |    .    .                 .          .          *  .
+            |    .    .                 .          .             *
+            |    .    .                 .          .                *
+            |    .    .                 .          .                   *                                
+            |----|----|-----------------|----------|---------------------->
+                 Nc   Nd                Nl         Nk
+                             Endurance
+
+
+    The slope parameters (m1, m2, m3, m4), endurance value at the knee points
+    of the quadlinear curve (Nd, Nl, Nk) and the `detail category` or
+    `characteristic` stress and endurance (Sc, Nc) is used to define
+    the quadlinear curve. 
+
+    Example
+    -------
+    >>> import fatpack
+    >>> import numpy as np
+    >>> np.random.seed(10)
+
+    First we create an endurance curve with detail category 90.
+    >>> curve = fatpack.BiLinearEnduranceCurve(90.)
+
+    Let us find the damage according to Miner's linear damage rule from
+    a rainflow counted signal y
+
+    >>> y = np.random.normal(size=100000) * 10.
+    >>> S = fatpack.find_rainflow_ranges(y)
+    >>> D = curve.find_miner_sum(S)
+    >>> print("The damage in signal y is D={0:3.2e}".format(D))
+    The damage in signal y is D=1.19e-04
```

Comment:
Please update the example such that the result can be checked with the code (so you should be able to reproduce the outcome).